plugins/ingest/src/test/java/org/elasticsearch/ingest/processor/AbstractStringProcessorTestCase.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.ingest.processor;
+
+import org.elasticsearch.ingest.Data;
+import org.elasticsearch.ingest.RandomDataPicks;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.*;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public abstract class AbstractStringProcessorTestCase extends ESTestCase {
+
+    protected abstract AbstractStringProcessor newProcessor(Collection<String> fields);
+
+    protected String modifyInput(String input) {
+        return input;
+    }
+
+    protected abstract String expectedResult(String input);
+
+    public void testProcessor() throws IOException {
+        Data data = RandomDataPicks.randomData(random());
+        int numFields = randomIntBetween(1, 5);
+        Map<String, String> expected = new HashMap<>();
+        for (int i = 0; i < numFields; i++) {
+            String fieldValue = randomAsciiOfLengthBetween(1, 30);
+            String fieldName = RandomDataPicks.addRandomField(random(), data, modifyInput(fieldValue));
+            expected.put(fieldName, expectedResult(fieldValue));
+        }
+        Processor processor = newProcessor(expected.keySet());
+        processor.execute(data);
+        for (Map.Entry<String, String> entry : expected.entrySet()) {
+            assertThat(data.getPropertyValue(entry.getKey(), String.class), equalTo(entry.getValue()));
+        }
+    }
+
+    public void testLowercaseNullValue() throws IOException {
```

Comment:
maybe omit lowercase from the method names here? (since these tests also run for uppercase and trim)